source/ui/x11/X11Window.ooc

```diff
@@ -52,8 +64,34 @@ X11Window: class extends NativeWindow {
 	}
 	resize: func (size: IntSize2D) { XResizeWindow(this display, this backend, size width, size height) }
 	free: override func {
+		if (this _xImage)
+			XDestroyImage(this _xImage)
+		XFreeGC(this display, this _x11GraphicsContext)
+		XDestroyWindow(this display, this backend)
 		XCloseDisplay(this display)
 		super()
 	}
+	draw: func (image: RasterYuv420Semiplanar) {
```

Comment:
This format is chosen only because this is the default one when running without gpu. Could be more like *any format that can return color at (x,y)*, but I'm too afraid of generics in ooc.